### PR TITLE
For #306: Implementing reservation Settings views

### PR DESCRIPTION
**Pull request by paulodamaso**
For #306: Implementing reservation Settings views according to generic views.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#451](https://github.com/timelesslounge/timelessis/issues/451) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#451](https://github.com/timelesslounge/timelessis/issues/451) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/6633dac9-0eab-4b2f-86fc-2dea15d8eaae); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/reservations/views.py [hunk 1]
```diff
@@ -22,34 +23,44 @@ class SettingsList(views.ListView):
...
-#     model = models.ReservationSettings
-#     template_name = "restaurants/tables/create_edit.html"
-#     success_url_name = "reservation_settings_list"
-#     not_found_url_name = "reservation_settings_list"
+class SettingsDetailView(views.DetailView):
+    """ Detail view for Reservation Settings  """
+    model = models.ReservationSettings
+    template_name = "restaurants/tables/create_edit.html"
+    success_view_name = "reservation_settings_list"
+    form_class = ReservationForm
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @paulodamaso `DetailView` doesn work with `success_view_name` and `form_class`. `DetailView` only pass object to template context without form.

File: timeless/reservations/views.py [hunk 2]
```diff
@@ -22,34 +23,44 @@ class SettingsList(views.ListView):
...
+
+
+SettingsCreateView.register(BP, "/settings/edit/")
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @paulodamaso I guess here should be 
`SettingsUpdateView.register(BP, "/settings/edit/<int:setting_id>")`, not `SettingsCreateView` you used

File: timeless/reservations/views.py [hunk 3]
```diff
@@ -22,34 +23,44 @@ class SettingsList(views.ListView):
...
+
+
+class SettingsDelete(views.DeleteView):
+    """ Delete view for Reservation Settings  """
+    model = models.ReservationSettings
+    success_view_name = "reservation_settings_list"
+    form_class = ReservationForm
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @paulodamaso `form_class` is not needed for `DeleteView`. it just deletes an object from db by id without using form

File: timeless/reservations/views.py [hunk 4]
```diff
@@ -22,34 +23,44 @@ class SettingsList(views.ListView):
...
 
 
-# class SettingsDeleteView(views.DeleteView):
-#     success_url_name = "reservation_settings_list"
+SettingsCreateView.register(BP, "/settings/delete/")
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @paulodamaso  
```python
SettingsDelete.register(BP, "/settings/delete/<int:setting_id>")
```

**Comment by paulodamaso** [comment]
@vryazanov Thank you for the comments, I've fixed them all please take a look

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 13 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#451](https://github.com/timelesslounge/timelessis/issues/451) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#451](https://github.com/timelesslounge/timelessis/issues/451) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**